python/pyspark/sql/udf.py

```diff
@@ -26,6 +26,28 @@
 
 
 def _wrap_function(sc, func, returnType):
+    def coerce_to_str(v):
+        import datetime
+        if type(v) == datetime.date or type(v) == datetime.datetime:
+            return str(v)
```

Comment:
I think it's weird that we have a cast here alone ... Can't we register a custom Pyrolite unpickler? Does it make the things more complicated?